subprojects/core/src/main/java/org/gradle/initialization/buildsrc/BuildSrcBuildListenerFactory.java

```diff
@@ -61,9 +64,16 @@ public void projectsLoaded(Gradle gradle) {
 
         @Override
         public void onConfigure(GradleInternal gradle) {
-            BuildableJavaComponent mainComponent = mainComponentOf(gradle);
+            final BuildableJavaComponent mainComponent = mainComponentOf(gradle);
             gradle.getStartParameter().setTaskNames(mainComponent.getBuildTasks());
-            classpath = mainComponent.getRuntimeClasspath().getFiles();
+            ProjectState projectState = mainComponent.getProject().getMutationState();
```

Comment:
Unless I missed another instance, I think this is the only place that needs `BuildableJavaComponent.getProject`.  I think we don't need this and can assume it's the root project because that's what `mainComponentOf` already does.

Reply:
Yeah, makes sense - changed to just assume `rootProject`.